CONTRIBUTING.md

```diff
@@ -168,12 +168,58 @@ These are garage-specific rules which are not part of the aforementioned style g
     import dowel.tabular as tabular
     ```
 
+### Other languages
+Non-Python files (including XML, HTML, CSS, JS, and Shell Scripts) should follow the [Google Style Guide](https://github.com/google/styleguide) for that language
+
+YAML files should use 2 spaces for indentation.
+
+### Whitespace (all languages)
+* Use Unix-style line endings
+* Trim trailing whitespace from all lines
+* All files should end in a single newline
+
 ## Documentation
 Python files should provide docstrings for all public methods which follow [PEP257](https://www.python.org/dev/peps/pep-0257/) docstring conventions and [Google](http://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings) docstring formatting. A good docstring example can be found [here](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).
 
-Additional standards:
+### Docstrings
 * Docstrings for `__init__` should be included in the class docstring as suggested in the [Google example](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).
 * Docstrings should provide full type information for all arguments, return values, exceptions, etc. according to the Google format
+* When documenting fields which are numpy arrays or other tensor types (and collections thereof), please carefully document the expected input shape of the field. See below for shape conventions.
+* For shapes and equations, use the Sphinx `:math:` directive to render them properly with mathematical symbols. 
+
+#### Conventions for documenting tensor shapes
+Data which include a meaningful time-series dimension (e.g. trajectories) should always document that dimension explicitly, even if that dimension has been flattened out. Data containing only non time-series samples should omit the time dimension.
+
+Always use the Sphinx `:math:` directive to render your shapes properly.
+
+| Symbol    | Description                                                                                             |
+|-----------|---------------------------------------------------------------------------------------------------------|
+| `(...)`   | Tensor shapes are enclosed in parentheses, e.g a batch of `(N, S^*)` samples                            |
+| `N`       | Batch dimension (e.g. trajectories or samples)                                                          |
+| `T`       | Time dimension                                                                                          |
+| `S^*`     | Variadic parts of a tensor shape, which will be broadcast or ignored                                    |
+| `[.]`     | Variable-length dimensions are enclosed in square brackets, e.g. `[K]` if `K` is the dimension variable |
+| `\bullet` | Flattening operator, e.g. `(N \bullet T, K)` has a flattened first dimension of length `N * T`          |
```

Comment:
I think it will be helpful to point out that then length of `N \bullet [T]` will be `sum(len(T_i))`